src/blockdev.rs

```diff
@@ -46,10 +46,10 @@ pub struct Disk {
 }
 
 impl Disk {
-    pub fn new(path: &str) -> Self {
-        Disk {
-            path: path.to_string(),
-        }
+    pub fn new(path: &str) -> Result<Self> {
+        Ok(Disk {
+            path: canonicalize(path)?.to_string_lossy().to_string(),
```

Comment:
Let's add a context here to make error messages more helpful? E.g.

```suggestion
            path: canonicalize(path)
                .with_context(|| format!("canonicalizing {}", path))?
                .to_string_lossy().to_string(),
```

Also, I know we use `to_string_lossy()` in a bunch of places, though here for this it'd be better IMO if we just hard error if it's not valid UTF-8 so we don't get weirder errors down the line on paths which build on top of it. We do this already in some places (see e.g. `is_dm_device()`).

Reply:
Will add context.

And will remove lossy. I initially added it because I didn't want to change the function signature / handle utf-8 errors, but now that we return Result there's no point in ignoring utf-8 errors.